clickplc/driver.py

```diff
@@ -106,8 +129,10 @@ async def get(self, address: str | None = None) -> dict:
         else:
             start, end = address, None
         i = next(i for i, s in enumerate(start) if s.isdigit())
-        category, start_index = start[:i].lower(), int(start[i:])
-        end_index = None if end is None else int(end[i:])
+
+        category = start[:i].lower()
+        start_index = 0.5 if start[i:].lower() == "0u" else int(start[i:])
+        end_index = 0.5 if end is not None and end[i:].lower() == "0u" else None if end is None else int(end[i:])
```

Comment:
What is going on here??

Is there some attempt to read only half a register?  Use a bitmask in that case.